features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java

```diff
@@ -498,15 +496,4 @@ public void afterPropertiesSet() throws Exception {
     public void reloadDaemonConfig(final IEvent e) {
         DaemonTools.handleReloadEvent(e, RemotePollerd.NAME, (event) -> handleConfigurationChanged());
     }
-
-    @EventHandler(uei = EventConstants.POLLER_PACKAGE_LOCATION_ASSOCIATION_CHANGED_EVENT_UEI)
-    public void reloadDaemonConfigForLocation(final IEvent e) {
```

Comment:
We need something similar here, that reacts to APPLICATION_CREATED_EVENT_UEI, APPLICATION_DELETED_EVENT_UEI and APPLICATION_CHANGED_EVENT_UEI and creates / reschedule / cancels only the affected and not the entire set of polling jobs, e.g. handleConfigurationChangedForApplication().

Reply:
if I understood @fooker correctly this ticket was only about creating the events, not listening to them. Having said that I can ad this of course as well. @fooker what do you think?

Reply:
That's right. I'm currently refactoring this anyways. So no need to implement the receiving side (right now).

Reply:
Ok, makes sense.